libraries/AR_Motors/AP_MotorsUGV.cpp

```diff
@@ -208,8 +208,8 @@ void AP_MotorsUGV::set_steering(float steering, bool apply_scaling)
 // set throttle as a value from -100 to 100
 void AP_MotorsUGV::set_throttle(float throttle)
 {
-    // only allow setting throttle if armed
-    if (!hal.util->get_soft_armed()) {
+    // only allow setting throttle if armed and not emergency stopped
+    if (!hal.util->get_soft_armed() || SRV_Channels::get_emergency_stop()) {
```

Comment:
I'm not sure about this one. Not allowing to set just means that it gets stuck at the last value until `output` comes round and zeros it. So we remove the race from every loop in E-stop but not the first entry. Maybe we should just set `_throttle = 0.0;` before the return. No harm in zeroing in two places.

Reply:
It seems a bit weird to me to try and set it to one value but instead have it set to another value.. that seems more odd than not being able to set it.  .. and it's an existing issue.. but in practice I don't think it matters either way so if you'd like it to set _throttle = 0.0 I can add that.